src/cpp-common/include/cpp-common/HashMap.h

```diff
@@ -0,0 +1,34 @@
+#ifndef PROPOLIS_HASHMAP_H
+#define PROPOLIS_HASHMAP_H
+
+#include "IHashMap.h"
+#include <type_traits>
+
+template <typename Key, typename MappedType, uint16_t maxSize>
+class HashMap : public IHashMap<Key, MappedType, maxSize> {
+  public:
+    HashMap();
+    ~HashMap() override;
+
+    bool insert(const std::pair<Key, MappedType>& item) override;
+    bool get(Key key, MappedType& item) const override;
+    std::optional<MappedType> at(Key key) const override;
+    bool remove(Key key) override;
```

Comment:
Maybe we could have an iterator/foreach function since we have no way to get back the stored keys as of now, though that could be for another PR, maybe add a TODO

Reply:
Added a TODO